docs/testing.md

```diff
@@ -0,0 +1,26 @@
+## Testing APB Image
+
+### Why would we want to test the image?
+The reason that we would want to be able to integration test APB's is that they have underlying images that could have been changed and would potentially cause the APB to fail. This means that to test the APB we would like to give the author the ability to verify certain actions by calling actions with default params.
+
+### Design
+The base APB entrypoint will learn the test action. The test action will be a test harness script *(This script will run pre-defined values for now, this should be configurable in the future)*.
+
+To include the testing of an APB add to the layout and add files and example of the layout is 
+```bash
+my-apb/
+├── apb.yml
+├── Dockerfile
+├── playbooks/
+    └── provision.yml  
+└── roles/
+    └── ...
+└── test/
+    └── provision_default.yml # Will be the defaults that will be used for the provision request. The parameters that are required should have default values here.
```

Comment:
Not sure we need to require any additional structure here, my guess would be that a stereotypical test.yml would something like:

```yaml

- name: Load test variables
  hosts: localhost
  connection: local
  tasks:
  - name: load test variables (one possible way)
    include_vars: test_defaults.yml
  - name: set test variables (another possible way)
    set_fact:
      a: 1
      b: 2
      c: 3

- include: provision.yaml

- name: Verify service successfully provisioned
  hosts: localhost
  connection: local
  roles:
    - test_verify
```

With the exact specifics of how variables are loaded or the task is verified is left to the author, similar to how we just delegate the exact behavior/structure of provision/deprovision/bind/unbind.

This would require no change to the base image.